File: Problems/0028_Implement_strStr/Implement_strStr.cs
using System;

public class Solution {
	public int StrStr(string haystack, string needle)
	{
		if (needle.Length == 0) {
			return 0;
		}

		if (haystack.Length == 0) {
			return -1;
		}

		int i, j, n;
		
		for ( i = 0 ; i < haystack.Length; i++ ) {
			for ( n = i, j = 0; j < needle.Length; j++, n++ ) {
				if ( n >= haystack.Length ) {
					return -1;
				}

				if ( haystack[n] != needle[j] ) {
					break;
				}
			}
			
			// Console.WriteLine("i = " + i.ToString() + ", j = " + j.ToString() );

			if ( j == needle.Length ) {
				return i;
			}
		}
	
		return -1;
	}

	public void Main()
	{
		//Console.Write( StrStr("hello", "ll") );
		//Console.Write( StrStr("aaaaa", "bba") );
		Console.Write( StrStr("mississippi", "issip") );
	}
}

File: Problems/0028_Implement_strStr/Implement_strStr.c
#include <stdio.h>

int strStr(char* haystack, char* needle) {
	if (*needle == '\0') {
		return 0;
	}

	if (*haystack == '\0') {
		return -1;
	}

	int len1, len2;
	int i, j, n;
	
	len1 = strlen(haystack);
	len2 = strlen(needle);
	
	for ( i = 0 ; i < len1; i++ ) {
		for ( n = i, j = 0; j < len2; j++, n++ ) {
			if ( n >= len1 ) {
				return -1;
			}

			if ( haystack[n] != needle[j] ) {
				break;
			}
		}

		
		if ( j == len2 ) {
			return i;
		}
	}

	return -1;
}

void main()
{
	//printf("%d\n", StrStr("hello", "ll") );
	//printf("%d\n", StrStr("aaaaa", "bba") );
	printf("%d\n", StrStr("mississippi", "issip") );
}
